keda/templates/22-metrics-deployment.yaml

```
@@ -42,6 +42,7 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ .Values.serviceAccount.name }}
+      automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}     
```

Let's use this approach instead:
```suggestion
      automountServiceAccountToken: {{ .Values.serviceAccount.automount.name }}
```

---

@tomkerkhove Thank you for suggested change. Unfortunately this will not work as autmountServiceAccountToken is a boolean field. Because of security restrictions we need to set autmountServiceAccountToken in serviceAccount to false and in deployments to true.

---

Let's go with this then?
```suggestion
      automountServiceAccountToken: {{ .Values.serviceAccount.automount.enabled }}
```